### PR TITLE
Fix FP4 inference corruption issue in glm4.5-air model

### PR DESCRIPTION
**Pull request by Azure-Tang**
# Background
When using FP4 quantization with the glm4.5-air model, the inference outputs may contain corrupted/garbled text. #8814 
# Root Cause
The quantization operator in sglang used `torch.empty` to create tensors;
* During padding, `empty` could introduce random values;
* These random values were later converted into NaN when mapped to nvfp4 format;
* In glm4.5’s first MLP layer, the intermediate size is 10944, which under tp=2 triggers padding. This may caused NaN values to be introduced into the matrix and propagated into subsequent GEMM computations;
* Therefore, the issue is always triggered in glm4.5’s first layer. In other layers or other models, the same issue may also occur whenever irregular shapes cause padding during quantization.
# Fix
* Replace torch.empty with explicitly initialized tensors (e.g., torch.zeros);
* Ensure padding regions are zeroed out, preventing random values from being converted to NaN in nvfp4 format and propagated downstream.
# Validation
* After the fix, glm4.5-air FP4 inference no longer produces corrupted output;
* Multiple benchmarks ran successfully with correct results.

image

By @Azure-Tang and @qiuqiu10

### CODE REVIEW

File: sgl-kernel/python/sgl_kernel/gemm.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def bmm_fp8(
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
     if out is None:
-        out = torch.empty(
+        out = torch.zeros(
```
Comment (Alcanderian) [inline sgl-kernel/python/sgl_kernel/gemm.py]: I am not appreciate that introducing extra zeros kernel in model forward phase. Can we double check whether these zeros is necessary？

Comment (Azure-Tang) [reply sgl-kernel/python/sgl_kernel/gemm.py]: I see, I have updated the pr and minimized modifications.

**Comment by Azure-Tang** [comment]
Hi, I have double checked pr and minimized modifications. @Alcanderian

**Review by Alcanderian — APPROVED** [review]
